### PR TITLE
ANA-1754: update inline valuation calls to include report currency

### PR DESCRIPTION
**Pull request by fbn-seamus**
# Pull Request Checklist

- [x] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass
- [~] Raised the PR against the `develop` branch

# Description of the PR

Adding report currency to all calls to inline valuation that are requesting the key `Valuation/PvInReportCcy` to match up with changes tightening the validation and consistency around that currency specific key.

### CODE REVIEW

File: src/Lusid.Instruments.Examples/Utilities/DemoInstrumentBase.cs [hunk 1]
```diff
@@ -202,7 +202,8 @@ internal void CallLusidInlineValuationEndpoint(LusidInstrument instrument, Model
                 metrics: TestDataUtilities.ValuationSpec,
                 sort: new List<OrderBySpec> {new OrderBySpec(TestDataUtilities.ValuationDateKey, OrderBySpec.SortOrderEnum.Ascending)},
                 valuationSchedule: valuationSchedule,
-                instruments: instruments);
+                instruments: instruments,
+                reportCurrency:reportCurrency);
```
Comment (fbn-seamus) [inline src/Lusid.Instruments.Examples/Utilities/DemoInstrumentBase.cs:206]: space

File: src/Lusid.Instruments.Examples/Utilities/InstrumentExamples.cs [hunk 2]
```diff
@@ -58,7 +58,7 @@ public static LusidInstrument GetExampleInstrument(string instrumentName)
             };
         }
 
-        internal static LusidInstrument CreateExampleFxForward(bool isNdf = true)
+        internal static FxForward CreateExampleFxForward(bool isNdf = true)
```
Comment (fbn-seamus) [inline src/Lusid.Instruments.Examples/Utilities/InstrumentExamples.cs:61]: All the changes in this file just allow my to access more information on the instrument by knowing the specific type.

**Review by ppwfinbourne — APPROVED** [review]
Standard/agreed changes.

We need a reporting currency to be present as, if people request a valuation without it, for lookthrough and other requests it can be badly defined. This can lead to nonsensical results which would be a poor outcome for the client. 

This is a precautionary change to rule those out.

**Pull request merged**